app/clients/hedera/mirror-node/model/transaction.go

```diff
@@ -65,44 +72,104 @@ type (
+		Links        Pagination       `json:"links"`
+	}
+	NftTransaction struct {
+		TransactionID     string `json:"transaction_id"`      // The transaction ID of the transaction
+		Type              string `json:"type"`                // The type of transaction TOKENBURN, TOKEMINT, CRYPTOTRANSFER
+		SenderAccountID   string `json:"sender_account_id"`   // The account that sent the NFT
+		ReceiverAccountID string `json:"receiver_account_id"` // The account that received the NFT
+	}
+	Pagination struct {
+		Next string `json:"next"` // Hyperlink to the next page of results
+	}
 )
 
 // getIncomingAmountFor returns the amount that is credited to the specified
 // account for the given transaction
-func (t Transaction) getIncomingAmountFor(account string) (string, string, error) {
+func (t Transaction) getIncomingAmountFor(account string) (int64, string, error) {
 	for _, tr := range t.Transfers {
 		if tr.Account == account {
-			return strconv.Itoa(int(tr.Amount)), constants.Hbar, nil
+			return tr.Amount, constants.Hbar, nil
 		}
 	}
-	return "", "", errors.New("no incoming transfer found")
+	return 0, "", errors.New("no incoming transfer found")
 }
 
 // getIncomingTokenAmountFor returns the token amount that is credited to the specified
 // account for the given transaction
-func (t Transaction) getIncomingTokenAmountFor(account string) (string, string, error) {
+func (t Transaction) getIncomingTokenAmountFor(account string) (int64, string, error) {
 	for _, tr := range t.TokenTransfers {
 		if tr.Account == account {
-			return strconv.Itoa(int(tr.Amount)), tr.Token, nil
+			return tr.Amount, tr.Token, nil
+		}
+	}
+	return 0, "", errors.New("no incoming token transfer found")
+}
+
+func (t Transaction) getIncomingNftTransferFor(account string) (serialNum int64, token string, err error) {
+	for _, ntr := range t.NftTransfers {
+		if ntr.ReceiverAccountID == account {
+			return ntr.SerialNumber, ntr.Token, nil
+		}
+	}
+
+	return 0, "", errors.New("no incoming nft transfer found")
+}
+
+// GetHBARTransfer gets the HBAR transfer for an Account
+func (t Transaction) GetHBARTransfer(account string) (amount int64, isFound bool) {
+	for _, tr := range t.Transfers {
+		if tr.Account == account {
+			return tr.Amount, true
 		}
 	}
-	return "", "", errors.New("no incoming token transfer found")
+
+	return 0, false
 }
 
-// GetIncomingTransfer returns the token amount OR the hbar amount that is credited to the specified
-// account for the given transaction. It depends on getIncomingAmountFor() and getIncomingTokenAmountFor()
-func (t Transaction) GetIncomingTransfer(account string) (string, string, error) {
+// GetIncomingTransfer returns the transfer to an account in the following order:
+// 1. Checks if there is an NFT transfer
+// 2. Checks if there is a Fungible Token transfer
+// 3. Checks if there is an HBAR transfer
+func (t Transaction) GetIncomingTransfer(account string) (isNft bool, amountOrSerialNum int64, asset string, err error) {
```

Comment:
The return result could be packed as struct type (in ex: TrasnferResponse), as otherwise, it is too lengthy for the return values.